authorization-server/src/modules/auth/guards/valid-refresh-token.guard.ts

```diff
@@ -9,12 +16,41 @@ export class ValidRefreshTokenGuard implements CanActivate {
   ) {}
 
   async canActivate(context: ExecutionContext): Promise<boolean> {
-    const { body, cookies } = context.switchToHttp().getRequest();
+    const request: Request = context.switchToHttp().getRequest();
+    const { body, cookies, method } = request;
+
+    let refreshToken: string | undefined;
+
+    if (method === 'POST') {
+      refreshToken = this.extractFromPostBodyOrCookies(body, cookies);
+    } else if (method === 'GET') {
+      refreshToken = this.extractFromQueryStringOrCookies(
+        request.cookies,
```

Comment:
If destructing above at the start of the function, might as well use it here.
```suggestion
        cookies,
```